### ISSUE TITLE
Add support for writing repeated bytes into a byte buffer.

### ISSUE THREAD

**Issue by Lukasa** [apple/swift-nio#1495]
I was working on a unit test where I needed `2**24` bytes in a byte buffer. Writing this the naive way is: `buffer.writeBytes(repeatElement(0, count: 1 << 24))`. This took around 16s. I replaced this with an unsafe pointer call to `memset` and it took more like 8ms.

We could solve this with a `writeRepeatedByte(_:count:)` that hides the unsafe code.

**Comment by shekhar-rajak** [comment]
Hi,

Just want to clarify. So the task here is to replace the `buffer.writeBytes(repeatElement(0, count: size))`  to `writeRepeatedByte(byte:0, count:size)` in test files, wherever it is present ?

**Comment by Lukasa** [comment]
Nope, the task is to do what @Davidde94 is doing in #1500. That said, a follow-up task once #1500 is merged could be to replace the use of that `writeBytes` pattern.

**Comment by shekhar-rajak** [comment]
Okay, Got it. Thanks!

**Comment by Davidde94** [comment]
I think it makes sense to replace all uses of `memset` / repeating bytes inside one PR
